core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -174,8 +174,8 @@ sealed abstract class Parser[+A] {
     * one to pick up after any error, resetting any state that was
     * modified by the left parser.
     */
-  def orElse[A1 >: A](that: Parser[A1]): Parser[A1] =
-    Parser.oneOf(this :: that :: Nil)
+  def orElse0[A1 >: A](that: Parser0[A1]): Parser0[A1] =
```

Comment:
I was just thinking... could we rename this to just `orElse`? wdyt?

Reply:
The result will be that `Parser` has this inherited `orElse`, and an overloaded `orElse` with `Parser` as an argument and returning a `Parser`. After making the change and search-and-replacing orElse0 with orElse, all the tests still pass.

So I think it's safe to say yes, we could.

I have no strong opinion either way on whether we should and whether we should in this particular PR.

The tension is between the convenience of just being able to say `.orElse` and getting the most specific result possible depending on the receiver and argument types and the obviousness of which method is called when one has the 0.

cc @johnynek